yurt/yurt_core/deploy.py

```diff
@@ -9,26 +17,98 @@
+
+def output_git_user_access_key():
+    cfg_parser = ConfigParser()
+    with open(os.path.expanduser('~/.yurtrc'), 'r') as yurtrc_file:
+        cfg_parser.readfp(yurtrc_file)
+        github_access_token = cfg_parser.get('Yurt', 'github_access_key')
+    return github_access_token
+
+def output_var_from_base_yml(variable):
+    project_dir = find_project_folder()
+    if project_dir:
+        base_vars_path = os.path.join(project_dir, 'orchestration', 'env_vars', 'base.yml')
+        with open(base_vars_path, 'r') as base_yaml_file:
+            base_yaml = yaml.load(base_yaml_file.read())
+            target_value = base_yaml.get(variable, None)
+        return target_value
+    else:
+        raise OSError('Not in a Yurt project folder')
+
 
 @deploy_cli.command()
 def deploy():
     """
     Starts deploy process
-    :return:
     """
     ENVIRONMENT_OPTIONS = {}
     if not os.path.exists("orchestration"):
-        print("You aren't in a Yurt repository yet. Use `cd` to get in the right directory.")
-        return
+        project_path = find_project_folder()
+        if project_path:
+            os.chdir(project_path)
+        else:
+            raise OSError('Not in a Yurt project folder')
 
     print("Option\tInventory")
     for index, filename in enumerate(os.listdir("./orchestration/inventory")):
         if not(filename == "vagrant"):
             ENVIRONMENT_OPTIONS[str(index)] = filename
             print("{0}:\t{1}".format(index, filename))
-    num_selection = raw_input("Which environment do you want to deploy (use Option number)?:\t")
+    num_selection = raw_input_wrapper("Which environment do you want to deploy (use Option number)?:\t")
     selection = ENVIRONMENT_OPTIONS[num_selection]
-    run('ansible-playbook -i orchestration/inventory/{0} orchestration/site.yml'.format(selection))
+
+    # Check if user has added deploy keys for the git repo
+    not_first_timer = raw_input_wrapper("Have you added deploy keys for this git repo yet (y/n)?: ", True)
+
+    if not_first_timer == 'n':
+        if not os.path.exists(os.path.expanduser('~/.yurtrc')):
+            # Create the access_key/yurtrc file
+            create_yurtrc_with_access_token()
+        # Use the access_key in ~/.yurtrc to POST git ssh pub key to server
+        access_key = output_git_user_access_key()
+        # Get needed variables from base_yml
+        git_ssh_pub_key = output_var_from_base_yml('git_ssh_pub_key')
+        git_repo = output_var_from_base_yml('git_repo')
+        project_name = get_project_name_from_repo(git_repo, drop_hyphens=False)
+        owner_name = get_owner_name_from_repo(git_repo)
+        compiled_curl_command = ' '.join((
+            'curl -X POST',
+            '-H "Content-Type:application/json"',
+            '-d \'{{"title": "{}@{}", "key": "{}"}}\''.format(project_name, selection, git_ssh_pub_key),
+            'https://api.github.com/repos/{}/{}/keys?access_token={}'.format(owner_name, project_name, access_key))),
```

Comment:
`owner` might be a misnomer. It's assumed that the person deploying has write access to the repo, yes.